gems/smithy/lib/smithy/anvil/client/views/shapes.rb

```diff
@@ -0,0 +1,254 @@
+              version: @service_shape.values.first['version']
+            )
+          end
+
+          def shapes_with_members
+            @shapes.select do |s|
+              %w[EnumShape IntEnumShape ListShape MapShape StructureShape UnionShape]
+                .include?(s.type)
+            end
+          end
+
+          def shapes
+            @shapes =
+              @service_index
+              .shapes_for(@service_shape)
+              .each_with_object([]) do |(k, v), arr|
+                next if %w[operation resource service].include?(v['type'])
+
+                arr << build_shape(k, v)
+              end
+          end
+
+          private
+
+          def build_operation_shape(id, shape)
+            OperationShape.new(
+              id: id,
+              name: build_shape_name(id).underscore,
+              input: build_shape_name(shape['input']['target']),
+              output: build_shape_name(shape['output']['target']),
+              errors: build_error_shapes(shape['errors']),
+              traits: filter_traits(shape['traits'])
+            )
+          end
+
+          def build_error_shapes(errors)
+            return [] if errors.nil?
+
+            errors.each_with_object([]) do |err, a|
+              a << build_shape_name(err['target'])
+            end
+          end
+
+          def build_shape(id, shape)
+            Shape.new(
+              id: id,
+              name: Vise::Shape.relative_id(id),
+              type: shape_type(shape['type']),
+              traits: filter_traits(shape['traits']),
+              members: build_member_shapes(shape)
+            )
+          end
+
+          def build_member_shapes(shape)
+            case shape['type']
+            when 'enum', 'intEnum', 'structure', 'union'
+              build_members(shape)
+            when 'list'
+              build_list_member(shape)
+            when 'map'
+              build_map_members(shape)
+            else
+              []
+            end
+          end
+
+          def build_members(shape)
+            shape['members'].each_with_object([]) do |(k, v), arr|
+              arr << build_member_shape(k, v['target'], v['traits'])
+            end
+          end
+
+          def build_list_member(shape)
+            m_shape = shape['member']
+            [] << build_member_shape('member', m_shape['target'], m_shape['traits'])
+          end
+
+          def build_map_members(shape)
+            %w[key value].map do |m_name|
+              m_shape = shape[m_name]
+              build_member_shape(m_name, m_shape['target'], m_shape['traits'])
+            end
+          end
+
+          def build_member_shape(name, shape, traits)
+            MemberShape.new(
+              name: name.underscore,
+              shape: build_shape_name(shape),
+              traits: filter_traits(traits)
+            )
+          end
+
+          def build_shape_name(id)
+            if PRELUDE_SHAPES_MAP.include?(id)
```

Comment:
Doesn't Vise::Shape.relative_id take care of prelude shapes in the same way? It should just remove the namespace i.e. `smithy.api#BigInteger` should be BigInteger.